File: 0260-single-number-iii/0260-single-number-iii.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        xor_sum=0
        for num in nums:
            xor_sum ^=num

        rightmost_set_bit=xor_sum & -xor_sum
            
        num1=0
        num2=0
        for num in nums:
            if num & rightmost_set_bit:
                num1^=num
            else:
                num2^=num

        return [num1,num2]
        

File: 0260-single-number-iii/0260-single-number-iii.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        map<int,int>cnt;
       vector<int>v;
        for(auto u:nums){
            cnt[u]++;
        }
        for(auto u:cnt){
            if(u.second==1){
                v.push_back(u.first);
            }
        }
        return v;
    }
};